BBcodeBundle/Parser/BBcodeParser.php

```diff
@@ -97,6 +92,64 @@ public function loadDefinitionsFromService(BBcodeDefinitionCollectionInterface $
      */
     public function getCodes()
     {
-        return $this->codes;
+        return $this->bbcodes;
+    }
+
+    /**
+     * This is the next step in parsing a tag. It's possible for it to still be invalid at this
+     * point but many of the basic invalid tag name conditions have already been handled.
+     *
+     * @param ElementNode $parent  the current parent element
+     * @param Tokenizer   $tokenizer  the tokenizer we're using
+     * @param string      $tagContent  the text between the [ and the ], assuming there is actually a ]
+     *
+     * @return ElementNode the new parent element
+     */
+    protected function parseTag(ElementNode $parent, Tokenizer $tokenizer, $tagContent)
+    {
+        $next;
+        if (!$tokenizer->hasNext() || ($next = $tokenizer->next()) != ']') {
+            $this->createTextNode($parent, '[');
+            $this->createTextNode($parent, $tagContent);
+            return $parent;
+        }
+
+        list($tmpTagName, $options) = $this->parseOptions($tagContent);
+
+        $actualTagName;
+        if ('' != $tmpTagName && '/' == $tmpTagName[0]) {
+           $actualTagName = substr($tmpTagName, 1);
+        } else {
+            $actualTagName = $tmpTagName;
+        }
+
+        if ('' != $tmpTagName && '/' == $tmpTagName[0]) {
+            $elToClose = $parent->closestParentOfType($actualTagName);
+            if (null == $elToClose || count($options) > 1) {
```

Comment:
What you see here is the original jbbcode. I have no other choice than copy/paste it to be able to inject our BBcodeBundle class on line 145 instead of the standard JBBcode one. So I think it's not carefull to patch the way it works, it may have unexpected behaviors.

Instead if you have another solution to avoid that copy/paste, I'm ok with it :)